crates/pessimistic-proof/src/aggchain_proof.rs

```diff
@@ -0,0 +1,36 @@
+use agglayer_primitives::Signature;
+pub use pessimistic_proof_core::aggchain_proof::{
+    AggchainProofData, AggchainProofECDSAData, AggchainProofPublicValues, AggchainProofSP1Data,
+    AggchainType, Vkey,
+};
+use pessimistic_proof_core::keccak::digest::Digest;
+use serde::{Deserialize, Serialize};
+
+// Aggchain proof values submitted via the [`Certificate`].
+#[derive(Serialize, Deserialize, Clone, Debug)]
+#[serde(untagged)]
+pub enum AggchainProof {
+    ECDSA { signature: Signature },
+    SP1 { aggchain_proof: AggchainProofSP1 },
+}
+
+impl AggchainProof {
+    pub fn aggchain_type(&self) -> AggchainType {
+        match self {
+            AggchainProof::ECDSA { signature: _ } => AggchainType::ECDSA,
+            AggchainProof::SP1 { aggchain_proof: _ } => AggchainType::SP1,
```

Comment:
This could be a `From` impl, but you can also update that like that:

```suggestion
            AggchainProof::ECDSA { .. } => AggchainType::ECDSA,
            AggchainProof::SP1 { .. } => AggchainType::SP1,
```

Reply:
I ended up deleted it in the end as it was not used